libs/barista-components/filter-field/src/filter-field.ts

```diff
@@ -1389,7 +1422,20 @@ export class DtFilterField<T = any>
 
   /** Updates the list of options or groups displayed in the autocomplete overlay */
   private _updateAutocompleteOptionsOrGroups(): void {
-    const currentDef = this._currentDef;
+    // In order to only display non defaultSearchDef's by filtering, we need to clone deep here.
+    const currentDef = cloneDeep(this._currentDef);
```

Comment:
I am not sure if the filtering here is correct. I think we can revert this whole function back. Let's hopp on a call